### ISSUE TITLE
visual: cursor may jump to wrong target

### ISSUE THREAD

**Issue by ret2libc** [radare/radare2#4920]
`r2 r2r/bins/mach0/ls-osx-x86_64`
```
                   0x100001443      4885c0         test rax, rax
              ┌──< 0x100001446      740f           je 0x100001457
              │    0x100001448      803800         cmp byte [rax], 0
            ┌────< 0x10000144b      740a           je 0x100001457
            │ │    0x10000144d      4889c7         mov rdi, rax
            │ │    0x100001450      e80f310000     call sym.imp.atoi
          ┌──────< 0x100001455      eb22           jmp 0x100001479
          │ └─└──> 0x100001457      488d55d0       lea rdx, [rbp - 0x30]
          │        0x10000145b      bf01000000     mov edi, 1
          │        0x100001460      be68740840     mov esi, 0x40087468
          │        0x100001465      30c0           xor al, al
          │        0x100001467      e882310000     call sym.imp.ioctl
          │        0x10000146c      83f8ff         cmp eax, -1
          │   ┌──< 0x10000146f      740e           je 0x10000147f
          │   │    0x100001471      0fb745d2       movzx eax, word [rbp - 0x2e]
          │   │    0x100001475      85c0           test eax, eax
          │ ┌────< 0x100001477      7406           je 0x10000147f
          └──────> 0x100001479      8905113e0000   mov dword [section.9.__data], eax ; [0x100005290:4]=80 LEA section.9.__data ; "P" @ 0x100005290
            └─└──> 0x10000147f      c7057b410000.  mov dword [0x100005604], 1 ; [0x100005604:4]=0
                   0x100001489      31db           xor ebx, ebx
              ┌──< 0x10000148b      eb1f           jmp 0x1000014ac
              │    0x10000148d      488d3d9d3600.  lea rdi, str.COLUMNS ; 0x100004b31 ; str.COLUMNS ; "COLUMNS" @ 0x100004b31
              │    0x100001494      e825310000     call sym.imp.getenv
              │    0x100001499      4885c0         test rax, rax
            ┌────< 0x10000149c      740e           je 0x1000014ac
            │ │    0x10000149e      4889c7         mov rdi, rax
```
If you go in visual mode and have a window size big enough to show the instruction at 0x10000148b, but not the instruction at 0x1000014ac, when you use the cursor and you position it on the instruction at 0x10000148b, then press enter, the cursor will go at the instruction 0x1000014b1, not at 0x1000014ac.

**Comment by radare** [comment]
I also found that Vc places the cursor at offset 16 instead of 0

**Comment by ret2libc** [comment]
Really? Please check with the latest commits if it still happens.

**Comment by radare** [comment]
yes, still does .i tested it yesterday night, and there's a test from r2r that already checks this and fails

**Comment by ret2libc** [comment]
Yesterday? TOO OLD :P
I committed some things for that just some minutes ago

**Comment by radare** [comment]
sorry, iwas in the shower :P looks gone now :)


thanks!

**Comment by ret2libc** [comment]
should be fixed with 8306463ae2d8cc252e5894b005a1777717c6c772
